### PR TITLE
[All][MBL-13894] Mobile Verify Beta

### PR DESCRIPTION
**Pull request by Blackchai**

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020258&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                        chain.proceed(chain.request())
+                    }
+                }.build()
+
+            val mobileVerifyBetaEnabled = RemoteConfigUtils.getString(
+                    RemoteConfigParam.MOBILE_VERIFY_BETA_ENABLED)?.equals("true", ignoreCase = true)
+                    ?: false
+
+            // We only want to switch over to the beta mobile verify domain if the remote firebase config is true
+            val baseUrl = if(mobileVerifyBetaEnabled && domain?.contains(".beta.") == true) {
```
Comment (joehoag) [inline libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: I think that this is the wrong domain to run the check on.  "domain" is the domain being checked, not the domain that our app is using.

Comment (joehoag) [reply libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: I take it back -- this is what we need to do.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+object MobileVerifyAPI {
+
+    internal interface OAuthInterface {
+        @GET("mobile_verify.json")
+        fun mobileVerify(@Query(value = "domain", encoded = false) domain: String?, @Query("user_agent") userAgent: String?): Call<DomainVerificationResult?>
+    }
+
+    private fun getAuthenticationRetrofit(domain: String?) : Retrofit {
+            val userAgent = userAgent
```
Comment (joehoag) [inline libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: Sorry -- where is the right-hand-side userAgent defined?

Comment (joehoag) [reply libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: Ah -- ApiPrefs.  Tricky (and tough to read).

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+    @JvmStatic
+    fun mobileVerify(domain: String?, callback: StatusCallback<DomainVerificationResult?>) {
+        if (paramIsNull(callback, domain)) {
+            return
+        }
+        val userAgent = userAgent
+        if (userAgent == "") {
+            d("User agent must be set for this API to work correctly!")
```
Comment (joehoag) [inline libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: Should we throw here?  Is it OK to move forward without actually making the oAuth call?

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt (hunk 2 above)
Comment (joehoag) [reply libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: How does ensuing logic disambiguate between this variable and ApiPrefs.userAgent?  They are both the same, so maybe it doesn't matter.

But I think that we should refer to the ApiPrefs version via ApiPrefs.userAgent, not an imported userAgent.

Ask: Let's not import ApiPrefs.userAgent.  Let's just call "ApiPrefs.userAgent" when we want to access that.

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt (hunk 3 above)
Comment (Blackchai) [reply libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: This is how this code has been for a long time, so I'm very hesitant to change the actual behavior. It looks new but its just converted from java.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt (hunk 2 above)
Comment (Blackchai) [reply libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: Changed it, I agree its more clear w/o the import. I'm surprised, that was just the converters default behavior.

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt (hunk 3 above)
Comment (joehoag) [reply libs/login-api-2/src/main/java/com/instructure/loginapi/login/api/MobileVerifyAPI.kt]: 👍 You're right.  I missed it the first time through, but this does mimic our old behavior.

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020274&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020275&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020276&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**